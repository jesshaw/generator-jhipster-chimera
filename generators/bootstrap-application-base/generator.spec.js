import { beforeAll, describe, expect, it } from 'vitest';

import { defaultHelpers as helpers, result } from 'generator-jhipster/testing';

const SUB_GENERATOR = 'bootstrap-application-base';
const BLUEPRINT_NAMESPACE = `jhipster:${SUB_GENERATOR}`;

describe('SubGenerator bootstrap-application-base of chimera JHipster blueprint', () => {
  describe('run', () => {
    beforeAll(async function () {
      await helpers
        .run(BLUEPRINT_NAMESPACE)
        .withJHipsterConfig()
        .withOptions({
          ignoreNeedlesError: true,
          blueprint: 'chimera',
        })
        .withJHipsterLookup()
        .withParentBlueprintLookup();
    });

    it('should succeed', () => {
      expect(result.getStateSnapshot()).toMatchSnapshot();
    });
  });
});
