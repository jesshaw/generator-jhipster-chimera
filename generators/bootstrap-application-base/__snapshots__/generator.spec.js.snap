// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SubGenerator bootstrap-application-base of chimera JHipster blueprint > run > should succeed 1`] = `
{
  ".yo-rc.json": {
    "stateCleared": "modified",
  },
}
`;
